pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -444,13 +444,20 @@ public void close() throws PulsarServerException {
     public CompletableFuture<Void> closeAsync() {
         mutex.lock();
         try {
+            // Close protocol handler before unloading namespace bundles because protocol handlers might maintain
+            // Pulsar clients that could send lookup requests that affect unloading.
+            if (protocolHandlers != null) {
+                protocolHandlers.close();
+                protocolHandlers = null;
+            }
             if (closeFuture != null) {
                 return closeFuture;
             }
             LOG.info("Closing PulsarService");
             if (brokerService != null) {
                 brokerService.unloadNamespaceBundlesGracefully();
             }
+            // It only tells the Pulsar clients that this service is not ready to serve for the lookup requests
```

Comment:
I wonder if `state = State.Closing` should be set before calling `brokerService.unloadNamespaceBundlesGracefully()`. /cc @heesung-sn @poorbarcode 
I guess extensible load manager prevents ownership assignment back to the same broker that is shutting down, but does the default modular load manager handle that similarly?

Reply:
I explained the order in https://github.com/apache/pulsar/pull/22573 but I cannot reproduce the stuck issue for now (not sure why `ExtensibleLoadManagerCloseTest` cannot prevent the order change).

Reply:
From the code analysis, `ServiceUnitStateChannelImpl#overrideOwnership` will send a message to the state topic (`loadbalancer-service-unit-state`). The call stack is

```
ServiceUnitStateChannelImpl#doCleanup
ServiceUnitStateChannelImpl#cleanOwnerships
ExtensibleLoadManagerImpl#disableBroker
...
BrokerService#unloadNamespaceBundlesGracefully
```

If the producer is disconnected, the reconnection and the followed lookup request will happen. In this case, if `state` is `Closing`, all lookup requests could fail and the send will fail with timeout.

Though since it does not wait for the send is done, it won't block the whole close process.